src/devices/Adxl345/Adxl345.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.Spi;
+using System.Runtime.InteropServices;
+using System.Device.Spi.Drivers;
+
+namespace Iot.Device.Adxl345
+{
+    /// <summary>
+    /// SPI Accelerometer ADX1345
+    /// </summary>
+    public class Adxl345 : IDisposable
+    {
+        private SpiDevice _sensor = null;
+
+        private readonly int _busId;
+        private readonly OSPlatform _os;
+        private readonly int _cs;                                
+        private readonly byte _gravityRangeByte;             
+        private readonly int _resolution = 1024;                      
+        private readonly int _range;
+
+        #region Addr
+        private const byte ADDRESS_POWER_CTL = 0x2D;        // Address of the Power Control register              
+        private const byte ADDRESS_DATA_FORMAT = 0x31;      // Address of the Data Format register               
+        private const byte ADDRESS_X0 = 0x32;               // Address of the X Axis data register                
+        private const byte ADDRESS_Y0 = 0x34;               // Address of the Y Axis data register              
+        private const byte ADDRESS_Z0 = 0x36;               // Address of the Z Axis data register               
+
+        private const byte ACCEL_SPI_RW_BIT = 0x80;         // Bit used in SPI transactions to indicate read/write  
+        private const byte ACCEL_SPI_MB_BIT = 0x40;         // Bit used to indicate multi-byte SPI transactions    
+        #endregion
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="os">The program runing platform (Linux or Windows10)</param>
+        /// <param name="busId">SPI Bus ID</param>
+        /// <param name="chipSelect">CS Port</param>
+        /// <param name="gravityRange">Gravity Range</param>
+        public Adxl345(OSPlatform os, int busId, int chipSelect, GravityRange gravityRange)
+        {
+            _busId = busId;
+            _os = os;
+            _cs = chipSelect;
+
+            if (gravityRange == GravityRange.Two)
+            {
+                _range = 4;
+            }
+            else if (gravityRange == GravityRange.Four)
+            {
+                _range = 8;
+            }
+            else if (gravityRange == GravityRange.Eight)
+            {
+                _range = 16;
+            }
+            else if (gravityRange == GravityRange.Sixteen)
+            {
+                _range = 32;
+            }
+            _gravityRangeByte = (byte)gravityRange;
+        }
+
+        /// <summary>
+        /// Initialize ADXL345
+        /// </summary>
+        public void Initialize()
```

Comment:
this feels like should be happening in the constructor (ok to have a separate private method)